hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/token/HapiTokenCancelAirdrop.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.spec.transactions.token;
+
+import static com.hedera.node.app.hapi.fees.usage.SingletonEstimatorUtils.ESTIMATOR_UTILS;
+import static com.hedera.services.bdd.spec.transactions.TxnUtils.suFrom;
+
+import com.hedera.node.app.hapi.fees.usage.TxnUsageEstimator;
+import com.hedera.node.app.hapi.fees.usage.token.TokenDissociateUsage;
+import com.hedera.node.app.hapi.utils.fee.SigValueObj;
+import com.hedera.services.bdd.spec.HapiSpec;
+import com.hedera.services.bdd.spec.transactions.HapiTxnOp;
+import com.hedera.services.bdd.spec.transactions.TxnUtils;
+import com.hederahashgraph.api.proto.java.FeeData;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.NftID;
+import com.hederahashgraph.api.proto.java.PendingAirdropId;
+import com.hederahashgraph.api.proto.java.TokenCancelAirdropTransactionBody;
+import com.hederahashgraph.api.proto.java.Transaction;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class HapiTokenCancelAirdrop extends HapiTxnOp<HapiTokenCancelAirdrop> {
+
+    private final List<Function<HapiSpec, PendingAirdropId>> pendingAirdropIds;
+
+    @SafeVarargs
+    public HapiTokenCancelAirdrop(Function<HapiSpec, PendingAirdropId>... pendingAirdropIds) {
+        this.pendingAirdropIds = List.of(pendingAirdropIds);
+    }
+
+    @Override
+    protected HapiTokenCancelAirdrop self() {
+        return null;
+    }
+
+    @Override
+    public HederaFunctionality type() {
+        return HederaFunctionality.TokenCancelAirdrop;
+    }
+
+    @Override
+    protected Consumer<TransactionBody.Builder> opBodyDef(HapiSpec spec) throws Throwable {
+        final var pendingAirdrops =
+                pendingAirdropIds.stream().map(f -> f.apply(spec)).toList();
+        final TokenCancelAirdropTransactionBody opBody = spec.txns()
+                .<TokenCancelAirdropTransactionBody, TokenCancelAirdropTransactionBody.Builder>body(
+                        TokenCancelAirdropTransactionBody.class, b -> b.addAllPendingAirdrops(pendingAirdrops));
+        return builder -> builder.setTokenCancelAirdrop(opBody);
+    }
+
+    public static Function<HapiSpec, PendingAirdropId> cancelPendingAirdrop(
```

Comment:
```suggestion
    public static Function<HapiSpec, PendingAirdropId> pending(
```

Reply:
Nit:
tokenCancelAirdrop(cancelPendingAirdrop(sender, receiver, token))) seems a little long to read, maybe we could exclude the cancel and Airdrop keywords as we already use them in tokenCancelAirdrop

Reply:
I agree for the `cancel` prefix but am not sure about the `Airdrop` postfix because in the case of 10 pending airdrops we might hide the `tokenCancelAirdrop` detail